src/core/atomic.d

```diff
@@ -161,34 +156,34 @@ version( CoreDdoc )
 else version( AsmX86_32 )
 {
     // Uses specialized asm for fast fetch and add operations
-    private HeadUnshared!(T) atomicFetchAdd(T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc
+    private HeadUnshared!(T) atomicFetchAdd(T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc @safe
         if( T.sizeof <= 4 && V1.sizeof <= 4)
     {
         size_t tmp = mod; // convert all operands to size_t
-        asm pure nothrow @nogc
+        asm pure nothrow @nogc @trusted
         {
             mov EAX, tmp;
             mov EDX, val;
         }
-        static if (T.sizeof == 1) asm pure nothrow @nogc { lock; xadd[EDX], AL; }
-        else static if (T.sizeof == 2) asm pure nothrow @nogc { lock; xadd[EDX], AX; }
-        else static if (T.sizeof == 4) asm pure nothrow @nogc { lock; xadd[EDX], EAX; }
+        static if (T.sizeof == 1) asm pure nothrow @nogc @trusted { lock; xadd[EDX], AL; }
+        else static if (T.sizeof == 2) asm pure nothrow @nogc @trusted { lock; xadd[EDX], AX; }
+        else static if (T.sizeof == 4) asm pure nothrow @nogc @trusted { lock; xadd[EDX], EAX; }
 
-        asm pure nothrow @nogc
+        asm pure nothrow @nogc @trusted
         {
             mov tmp, EAX;
         }
 
         return cast(T)tmp;
     }
 
-    private HeadUnshared!(T) atomicFetchSub(T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc
+    private HeadUnshared!(T) atomicFetchSub(T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc @safe
         if( T.sizeof <= 4 && V1.sizeof <= 4)
     {
         return atomicFetchAdd(val, -mod);
     }
 
-    HeadUnshared!(T) atomicOp(string op, T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc
+    HeadUnshared!(T) atomicOp(string op, T, V1)( ref shared T val, V1 mod ) pure nothrow @nogc @trusted
```

Comment:
As far as I can tell, the only thing done with `val` is take its address. No T operators.

There's a mistake in the design of `cas` in that it takes a pointer rather than a reference, this messes with safety checks. But fixing that is beyond the scope of this PR for now.